apps/src/pythonlab/pyodideWorkerManager.ts

```diff
@@ -285,15 +287,22 @@ const asyncRun = (() => {
 })();
 
 const restartPyodideIfProgramIsRunning = () => {
+  // Always send a stop message, as some programs will still
+  // look like they are "running" to the user even if they aren't truly running
+  // (for example, the neighborhood). We send via the console manager rather than
+  // the message handler because the neighborhood stops processing messages on stop,
+  // and we want to always show this to the user.
+  const consoleManager = CodebridgeRegistry.getInstance().getConsoleManager();
```

Comment:
Just checking that we're okay if the 'Program stopped.' message is on the same line as user output.

![Screenshot 2025-04-24 at 11 29 22 AM](https://github.com/user-attachments/assets/5fce3a91-c56d-4bdb-90c3-661d6b1bd3db)

Reply:
Also wondering about updating message to 'Program manually stopped'.
When a neighborhood project is run and the program stops after the animation is complete, would users  expect a message at that point?

cc: @samantha-code

Reply:
It should normally be on a new line, that example is because they were on an input line.

I will note we only show `Program stopped` when the user manually clicks the `Stop` button. I feel like "manually" isn't really a necessary addition since it's only when the user clicks stop?

Reply:
And we will show it whenever the user clicks "stop", neighborhood or not.

Reply:
Thanks @samantha-code !
> I wonder if thats because it was stopped on an input line?

Yes - that's correct. This happened when program manually stopped before user enters input in terminal. Updated my comment above.

Reply:
>I think the user would expect this message each time the program was manually stopped, regardless of the neighborhood or not.

Yes, this is how it's implemented currently. Thanks!

Reply:
>I will note we only show Program stopped when the user manually clicks the Stop button. I feel like "manually" isn't really a necessary addition since it's only when the user clicks stop?

Not a strong feeling, but I think adding 'manually' seems more accurate and may prevent a user wondering about why it doesn't print when a neighborhood program stops.

Reply:
It will print if the user presses stop on a neighborhood program